api/deviceadm/service.go

```diff
@@ -88,24 +88,30 @@ func (s *service) DeleteDevice(ctx context.Context, uid models.UID, tenant, user
 func (s *service) RenameDevice(ctx context.Context, uid models.UID, name, tenant, username string) error {
 	err := s.isNamespaceOnwer(ctx, tenant, username)
 	if err != nil {
-		return err
+		return ErrUnauthorized
+	}
+
+	device, err := s.store.DeviceGetByUID(ctx, uid, tenant)
+	if err != nil {
+		return err		// Not Found
 	}
 
-	device, _ := s.store.DeviceGetByUID(ctx, uid, tenant)
 	validate := validator.New()
 	name = strings.ToLower(name)
-	if device != nil {
-		if device.Name != name {
-			device.Name = name
-			if err := validate.Struct(device); err == nil {
-				otherDevice, _ := s.store.DeviceGetByName(ctx, name, tenant)
-				if otherDevice == nil {
-					return s.store.DeviceRename(ctx, uid, name)
-				}
-			}
-		}
+	if device.Name == name {
+		return			// I need return the "Error name is equal"
```

Comment:
return nil // No operation needed.

Reply:
Could i return a specific error stating that the name is the same?

Reply:
I think it does make sense. And who uses it, need to handle this as well.

Reply:
Yes, I will do this